lib/spack/spack/test/spec_semantics.py

```diff
@@ -243,6 +245,123 @@ def test_satisfies_matching_variant(self):
         check_satisfies('mpich~foo', 'mpich foo=FALSE')
         check_satisfies('mpich foo=False', 'mpich~foo')
 
+    def test_satisfies_multi_value_variant(self):
+        check_satisfies('multivalue_variant foo="bar,baz"',
```

Comment:
Variant parser shouldn't require quotes for this (but it's ok to use them).  Can we add a test without "" around the variant value?